zebra/rt_netlink.c

```diff
@@ -348,8 +348,41 @@ netlink_route_change_read_unicast (struct sockaddr_nl *snl, struct nlmsghdr *h,
         }
     }
   else
-    rib_delete (afi, SAFI_UNICAST, vrf_id, ZEBRA_ROUTE_KERNEL, 0, flags,
-                &p, NULL, gate, index, table);
+    {
+      if (!tb[RTA_MULTIPATH])
+        rib_delete (afi, SAFI_UNICAST, vrf_id, ZEBRA_ROUTE_KERNEL, 0, flags,
+                    &p, NULL, gate, index, table);
+      else
```

Comment:
The commit title says this is for v6, but it actually applies v4 and v6. The old version can only handle one next hop route delete regardless of it being v4 or v6. What this commit brings is support for deleting a route with multiple hops. The Commit comment is not clear on this either. I suggest to just change the commit message to reflect this.